### PR TITLE
Eskip tool

### PR DESCRIPTION
**Pull request by aryszka**
low prio, but can be a useful tool
there some smaller changes to the etcd client, otherwise doesn't affect the main code

### CODE REVIEW

File: cmd/eskip/doc.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+Delete routes from etcd:
+
+    eskip delete -ids route1,route2,route3
+
+Delete all routes from etcd:
+
+    eskip print | eskip delete
+
+Copy all routes in etcd to under another prefix:
```
Comment (mo-gr) [inline cmd/eskip/doc.go]: the under is misplaced

File: cmd/eskip/write.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	}
+
+	id, err := flowid.NewFlowId(randomIdLength)
+	if err != nil {
+		return err
+	}
+
+	// replace characters that are not allowed
+	// for eskip route ids.
+	id = routeIdRx.ReplaceAllString(id, "x")
```
Comment (mo-gr) [inline cmd/eskip/write.go]: why not just use a uuid?

File: etcd/etcd.go [hunk 3]
```diff
@@ -34,14 +37,27 @@ tests.
...
+// RouteInfo contains a route id, plus the loaded and parsed route or
+// the parse error in case of failure.
+type RouteInfo struct {
+
+	// The route id plus the route data or if parsing was successful.
+	eskip.Route
+
+	// The parsing error if the parsing failed.
+	ParseError error
+}
```
Comment (mo-gr) [inline etcd/etcd.go]: I guess this thing shouldn't contain both at the same time, right? Is there an `Either` type in go?

**Comment by mo-gr** [comment]
lgtm

**Comment by mo-gr** [comment]
And huge 👍🏻 for that awesome eskip tool!

File: cmd/eskip/write.go (hunk 2 above)
Comment (aryszka) [inline cmd/eskip/write.go]: I didn't want to add an additional dependency, and considered the randomness of this solution good enough for the purpose.

File: etcd/etcd.go (hunk 3 above)
Comment (aryszka) [inline etcd/etcd.go]: It shouldn't be used other than receiving what was stored here. It means that there is a route here, and it is either valid, or this is the problem with it. There is no such thing as `Either`, but true, this is used in a similar way.

File: cmd/eskip/doc.go (hunk 1 above)
Comment (aryszka) [inline cmd/eskip/doc.go]: fixed, thanks

**Pull request merged**